OpenRA.Mods.Common/Projectiles/LaserZap.cs

```diff
@@ -100,17 +103,18 @@ public void Tick(World world)
 
 		public IEnumerable<IRenderable> Render(WorldRenderer wr)
 		{
-			if (wr.World.FogObscures(target) &&
-				wr.World.FogObscures(args.Source))
+			if (!info.RenderBeam && hitanim == null)
 				yield break;
 
-			if (ticks < info.BeamDuration)
+			var targetObscured = wr.World.FogObscures(target);
+			var sourceObscured = wr.World.FogObscures(args.Source);
+			if (!(targetObscured && sourceObscured) && ticks < info.BeamDuration)
```

Comment:
This will cause the whole beam to be invisible if you get shot by an obelisk hidden under the fog, which looks much worse than what we have right now (I had already tried this a long time ago).